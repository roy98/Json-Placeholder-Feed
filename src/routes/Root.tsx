import React from 'react'
import MainLayout from '../components/layout/mainLayout'
import Posts from '../components/posts/Posts'

function Root() {

  return (
    <>
        <MainLayout>
            <Posts />
        </MainLayout>
    </>
  )
}

export default Root
