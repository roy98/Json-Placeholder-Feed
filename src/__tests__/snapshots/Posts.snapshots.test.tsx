import React from 'react'
import renderer from "react-test-renderer"
import { BrowserRouter } from 'react-router-dom'
import Root from "../../routes/Root"

test("renders root page that list all posts", () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Root />
    </BrowserRouter>
  ).toJSON();
  
  expect(tree).toMatchSnapshot();
});