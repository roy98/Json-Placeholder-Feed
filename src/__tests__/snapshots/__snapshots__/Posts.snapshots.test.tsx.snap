// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders root page that list all posts 1`] = `
<div
  id="container"
>
  <header>
    <nav>
      <h2>
        <a
          href="/"
          onClick={[Function]}
        >
          JSON Placeholder Feed
        </a>
      </h2>
    </nav>
  </header>
  <main>
    <div
      className="main-content"
    >
      <div>
        <div
          className="search-container"
        >
          <input
            data-testid="search-1"
            onInput={[Function]}
            onKeyDown={[Function]}
            placeholder="Search a post by title"
            title="Search by title"
            type="search"
            value=""
          />
          <a
            className="btn"
            onClick={[Function]}
          >
            Search
          </a>
        </div>
        <p
          data-testid="no-data-1"
          style={
            {
              "margin": "4rem",
              "textAlign": "center",
            }
          }
        >
          No Data found
        </p>
      </div>
    </div>
  </main>
  <footer>
    <strong>
      JSON Placeholder Feed
    </strong>
    <p>
      Â©Copyright 
      2023
    </p>
    <code>
      Test Powered by Digit-AD
    </code>
  </footer>
</div>
`;
